packages/schema/src/make-remote-executable-schema.ts

```diff
@@ -0,0 +1,179 @@
+  const queryType = schema.getQueryType();
+  const queries = queryType.getFields();
+  Object.keys(queries).forEach((key) => {
+    queryResolvers[key] = createResolver(fetcher);
+  });
+
+  // prepare mutation resolvers
+  const mutationResolvers: IResolverObject = {};
+  const mutationType = schema.getMutationType();
+  if (mutationType) {
+    const mutations = mutationType.getFields();
+    Object.keys(mutations).forEach((key) => {
+      mutationResolvers[key] = createResolver(fetcher);
+    });
+  }
+
+  // prepare subscription resolvers
+  const subscriptionType = schema.getSubscriptionType();
+  if (subscriptionType) {
+    throw new Error('Subscriptions are not yet implemented');
+  }
+
+  // merge resolvers into resolver map
+  const resolvers: IResolvers = { [queryType.name]: queryResolvers };
+
+  if (!isEmptyObject(mutationResolvers)) {
+    resolvers[mutationType.name] = mutationResolvers;
+  }
+
+  // add missing abstract resolvers (scalar, unions, interfaces)
+  const typeMap = schema.getTypeMap();
+  const types = Object.keys(typeMap).map((name) => typeMap[name]);
+
+  for (const type of types) {
+    if (type instanceof GraphQLInterfaceType || type instanceof GraphQLUnionType) {
+      resolvers[type.name] = {
+        __resolveType(parent: any, _context: any, info: any) {
+          const parentTypename: string = parent['__typename'];
+          if (!parentTypename) {
+            throw new Error('Did not fetch typename for object, unable to resolve interface.');
+          }
+
+          const resolvedType = info.schema.getType(parentTypename);
+
+          if (!(resolvedType instanceof GraphQLObjectType)) {
+            throw new Error('__typename did not match an object type: ' + parentTypename);
+          }
+
+          return resolvedType;
+        },
+      };
+    } else if (type instanceof GraphQLScalarType) {
+      if (
+        !(
+          type === GraphQLID ||
+          type === GraphQLString ||
+          type === GraphQLFloat ||
+          type === GraphQLBoolean ||
+          type === GraphQLInt
+        )
+      ) {
+        resolvers[type.name] = new GraphQLScalarType({
+          name: type.name,
+          description: type.description,
+          astNode: type.astNode,
+          serialize(value: any) {
+            return value;
+          },
+          parseValue(value: any) {
+            return value;
+          },
+          parseLiteral(ast: ValueNode) {
+            return parseLiteral(ast);
+          },
+        });
+      }
+    }
+  }
+
+  return makeExecutableSchema({
+    typeDefs,
+    resolvers,
+  });
+}
+
+export function createResolver(fetcher: Fetcher): GraphQLFieldResolver<any, any> {
+  return async (_root, _args, _context, info) => {
+    const fragments = Object.keys(info.fragments).map((fragment) => info.fragments[fragment]);
+    const document = {
+      kind: Kind.DOCUMENT,
+      definitions: [info.operation, ...fragments],
+    };
+
+    return fetcher({
```

Comment:
This will cause some issues later. Let’s say remote schema is merged with another schema. It will send the whole query to the remote schema as well so some weird errors might be thrown. And also this will send the query for each root field.
https://github.com/apollographql/graphql-tools/issues/1346#issuecomment-609559637
This issue looks similar but it is for links. (Links work in a different way)